test/permit/permit.test.js

```diff
@@ -0,0 +1,277 @@
+const {setupPermit} = require("./fixtures.ts");
+const ethers = require("ethers");
+const {BigNumber} = ethers;
+const {splitSignature} = require("ethers/lib/utils");
+const {findEvents} = require("../utils");
+const sigUtil = require("eth-sig-util");
+const {expect} = require('../chai-setup');
+const {data712} = require("./data712.ts");
+const {bufferToHex} = require("ethereumjs-util");
+
+const zeroAddress = ethers.constants.AddressZero;
+const TEST_AMOUNT = BigNumber.from(10).mul("1000000000000000000");
+
+describe("Permit", function () {
+  let setUp;
```

Comment:
Instead of declaring mutable variable. I think it is better to have the setup function being called for every test like here : https://github.com/thesandboxgame/sandbox-private-contracts/blob/efa2018166ac5c5eeeb412e79612a3ffe2a44553/test/sand/Sand.test.ts#L33

This ensure variable are not changed by mistake. and make it easier to read in my opinion.

I did not change test in https://github.com/thesandboxgame/sandbox-private-contracts/tree/hardhat/test/liquidityMining to reflect that as it was quite some tasks to convert all tests

but going forward I think it make sense

the setup function can return all user, contracts required

Reply:
note that wallet, nonce, deadline could be declared as const outside of the beforeEach

Reply:
Note that this is my opinion and it is true that in some case, the use of beforeEach is handy.
But I would try to minimize it to at max one variable like you did with setup.

Reply:
Yes - agreed, I will look at this.